eng/bomgenerator/src/main/java/com/azure/tools/bomgenerator/models/BOMReport.java

```diff
@@ -0,0 +1,112 @@
+        if (!dependencyConflicts.containsKey(dependencyWithConflict)) {
+            dependencyConflicts.put(dependencyWithConflict, new ArrayList<>());
+        }
+
+        dependencyConflicts.get(dependencyWithConflict).add(conflictingDependency);
+    }
+
+    public void generateReport() {
+        System.out.println("Starting to write HTML report (" + OffsetDateTime.now() + ")");
+
+        out("<!DOCTYPE html>");
+        out("<html>");
+        out("  <head>");
+        out("    <title>Dependency Issues Report</title>");
+        out("    <meta charset=\"UTF-8\"/>");
+        out("    <style>");
+
+        // write out CSS inline
+        try (
+            BufferedReader r = new BufferedReader(new InputStreamReader(BOMReport.class.getResourceAsStream("report.css")))) {
+            r.lines().forEach(line -> out("      " + line));
+        } catch (Exception e) {
+            // no-op
+            System.out.println("Can't find file: " + BOMReport.class.getResource("report.css"));
+        }
+
+        out("    </style>");
+        out("  </head>");
+        out("  <body>");
+
+        out("    <center>");
+
+        // Summary table
+        out("      <h1>Dependency Conflicts Report</h1>");
+        out("      <p>This report analyzed all track 2 GA data plane libraries and found the following conflicts in its dependency management.<br/>" +
+            "It is important to resolve these conflicts to ensure convergence of the dependencies.<br/></p>");
+
+        out("    <table>");
+        out("      <thead>");
+        out("<tr>");
+        out("<th>");
+        out("Dropped dependency");
+        out("</th>");
+        out("<th>");
+        out("Reasons");
+        out("</th>");
+        out("</tr>");
+        dependencyConflicts.keySet().stream().forEach(dependency -> {
+                out("<tr>");
+                out("<td>");
+                out(dependency.toString());
+                out("</td>");
+                out("<td>");
+                dependencyConflicts.get(dependency).forEach(conflict -> {
+                    out(String.format("Includes dependency %s, Expected dependency %s", conflict.getActualDependency(), conflict.getExpectedDependency()));
+                    out("<br/>");
+                });
+                out("</td>");
+                out("</tr>");
+        });
+        out("      <small>Report generated at " + DateTimeFormatter.ofPattern("HH:mm:ss 'on' yyyy-MM-dd").format(LocalDateTime.now()) + "<br/>");
+        out("    </center>");
+        out("  </body>");
+        out("</html>");
+
+        // write out to the output file
+        File outFile = new File(reportFileName);
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(outFile))) {
+            writer.write(sb.toString());
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        System.out.println("HTML report written to " + outFile + " (" + OffsetDateTime.now() + ")");
```

Comment:
Move this inside `try` block. This should be printed only if there's no exception when writing to file.